aws/resource_aws_glue_catalog_database.go

```diff
@@ -0,0 +1,180 @@
+		Create: resourceAwsGlueCatalogDatabaseCreate,
+		Read:   resourceAwsGlueCatalogDatabaseRead,
+		Update: resourceAwsGlueCatalogDatabaseUpdate,
+		Delete: resourceAwsGlueCatalogDatabaseDelete,
+		Exists: resourceAwsGlueCatalogDatabaseExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"catalog_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"location_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"parameters": {
+				Type:     schema.TypeMap,
+				Elem:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsGlueCatalogDatabaseCreate(d *schema.ResourceData, meta interface{}) error {
+	glueconn := meta.(*AWSClient).glueconn
+	catalogID := createAwsGlueCatalogID(d, meta)
+	name := d.Get("name").(string)
+
+	input := &glue.CreateDatabaseInput{
+		CatalogId: aws.String(catalogID),
+		DatabaseInput: &glue.DatabaseInput{
+			Name: aws.String(name),
+		},
+	}
+
+	_, err := glueconn.CreateDatabase(input)
+	if err != nil {
+		return fmt.Errorf("Error creating Catalogue Database: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s", catalogID, name))
+
+	return resourceAwsGlueCatalogDatabaseUpdate(d, meta)
+}
+
+func resourceAwsGlueCatalogDatabaseUpdate(d *schema.ResourceData, meta interface{}) error {
+	glueconn := meta.(*AWSClient).glueconn
+
+	catalogID, name := readAwsGlueCatalogID(d.Id())
+	input := &glue.UpdateDatabaseInput{
+		CatalogId: aws.String(catalogID),
+		DatabaseInput: &glue.DatabaseInput{
+			Name:        aws.String(name),
+			Description: aws.String(d.Get("description").(string)),
+			LocationUri: aws.String(d.Get("location_uri").(string)),
+			Parameters:  make(map[string]*string),
+		},
+		Name: aws.String(name),
+	}
+	for key, value := range d.Get("parameters").(map[string]interface{}) {
+		input.DatabaseInput.Parameters[key] = aws.String(value.(string))
+	}
+
+	doUpdate := d.HasChange("description")
+	doUpdate = doUpdate || d.HasChange("location_uri")
+	doUpdate = doUpdate || d.HasChange("parameters")
+
+	if doUpdate {
+		if _, err := glueconn.UpdateDatabase(input); err != nil {
+			return err
+		}
+	}
+
+	return resourceAwsGlueCatalogDatabaseRead(d, meta)
+}
+
+func resourceAwsGlueCatalogDatabaseRead(d *schema.ResourceData, meta interface{}) error {
+	glueconn := meta.(*AWSClient).glueconn
+
+	catalogID, name := readAwsGlueCatalogID(d.Id())
+
+	input := &glue.GetDatabaseInput{
+		CatalogId: aws.String(catalogID),
+		Name:      aws.String(name),
+	}
+
+	out, err := glueconn.GetDatabase(input)
+	if err != nil {
+		return fmt.Errorf("Error reading Glue Cataloge Database: %s", err.Error())
```

Comment:
nitpick: typo 👀

Reply:
Also do you mind adding a check for database which was removed outside of Terraform? something like this should suffice:

```go
if isAWSErr(err, glue.ErrCodeEntityNotFoundException, "") {
	log.Printf("[WARN] Glue Catalog Database (%s) not found, removing from state", d.Id())
	d.SetId("")
}
```
you may also add an acceptance test verifying this behaviour, similar to e.g. https://github.com/terraform-providers/terraform-provider-aws/blob/79ed212c594fd8e848d0274cd0efc70bbf4cede8/aws/resource_aws_eip_test.go#L220-L238

but the test is not a blocker.

Reply:
As I mentioned above, I wish to round out the whole AWS Glue product set for terraform. For the sake of my sanity (getting a PR merged), can I put this test in a subsequent PR?

Reply:
Adding a test in a subsequent PR is totally fine. 😉